### PR TITLE
Fix rollover bug

### LINKED ISSUE #743: The -R option doesn't work with -l option
#### Environment details (Put `x` in the checkbox along with the information)

[x] Operating System: Debian 10
[x] Desktop Environment: fvwm
[x] Terminal Emulator: xfce4-terminal
[x] Shell: bash
[ ] Custom desktop opener (if applicable):
[ ] Program options used:
[ ] Configuration options set:
[ ] Issue exists on `nnn` master

#### Exact steps to reproduce the issue
I found that the -R option only works when scrolling exactly to the last entry. For example, there are 17 files in a folder. run nnn -R -l 5, then I enter that folder, and use the mouse wheel to scroll. It will go to 1/17, 6/17, 11/17, 16/17, then rollover to 4/17,... 14/17, 2/17... and go on until 17/17.

### PR DESCRIPTION
**Pull request by leovilok**
Fixes #743

### CODE REVIEW

**Comment by leovilok** [comment]
Sorry for the branch created in `nnn` repo, I'm on a horrible network that block git pull/push, so I had to use the github web UI.

**Comment by leovilok** [comment]
And I realise tha maybe the expected behaviour is a little bit different:

- currently it stops scrolling if there is less space than what's provided in the `-l` option
- but one might expect to go to the top/end and get blocked here

What do you think ?

**Comment by jarun** [comment]
> but one might expect to go to the top/end and get blocked here

This.

**Comment by leovilok** [comment]
Done.

File: src/nnn.c [hunk 1]
```diff
@@ -5972,15 +5972,21 @@ static bool browse(char *ipath, const char *session, int pkey)
 			}
 #if NCURSES_MOUSE_VERSION > 1
 			/* Scroll up */
-			if (event.bstate == BUTTON4_PRESSED && ndents && (cfg.rollover || cur >= scroll_lines)) {
-				move_cursor((cur + ndents - scroll_lines) % ndents, 0);
+			if (event.bstate == BUTTON4_PRESSED && ndents && (cfg.rollover || cur)) {
+				if(!cfg.rollover && cur < scroll_lines)
```
Comment (jarun) [inline src/nnn.c:5976]: Space after `if`. Same at line 5986.

Is there no way to do it without an additional condition check?

@KlzXS ?

**Review by jarun — CHANGES_REQUESTED** [review]

File: src/nnn.c (hunk 1 above)
Comment (leovilok) [reply src/nnn.c]: Well you can hide it in a ternary operator, but it's really a special case, so we need to check it at some point.

**Pull request merged**